src/app/ReadHandler.h

```diff
@@ -0,0 +1,166 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *     This file defines read handler for a CHIP Interaction Data model
+ *
+ */
+
+#pragma once
+
+#include <app/InteractionModelDelegate.h>
+#include <app/reporting/EventLoggingTypes.h>
+#include <app/reporting/LoggingManagement.h>
+#include <core/CHIPCore.h>
+#include <core/CHIPTLVDebug.hpp>
+#include <messaging/ExchangeContext.h>
+#include <messaging/ExchangeMgr.h>
+#include <messaging/Flags.h>
+#include <protocols/Protocols.h>
+#include <support/CodeUtils.h>
+#include <support/DLLUtil.h>
+#include <support/logging/CHIPLogging.h>
+#include <system/SystemPacketBuffer.h>
+
+namespace chip {
+namespace app {
+namespace reporting {
+class ReportingEngine;
+}
+
+/**
+ *  @class ReadHandler
+ *
+ *  @brief The read handler is responsible for processing read request, asks reporting engine to pack report data and sending
+ * reports via this handler.
+ *
+ */
+class ReadHandler
+{
+private:
```

Comment:
Why are all ReadHandler methods all private?

A class with all private methods does not seem useful. It looks as a followup we are adding friend classes to actually make the class usable. Please make relevant usage methods public.